File: src/main/resources/assets/bonsaitrees/config/types.d/integrateddynamics.js
load("config/bonsaitrees/types.d/defaults.js");

isEnabled = function() {
    var Loader = Java.type("net.minecraftforge.fml.common.Loader");
    return Loader.isModLoaded("integrateddynamics");
}

var main = function(source) {
    var menrilTree = new TreeTypeSimple("integrateddynamics:menril", "integrateddynamics:menril_sapling", 0);
    menrilTree.setSource(source);
    menrilTree.addDrop("minecraft:stick", 0, stickAmount, stickChance);
    menrilTree.addDrop("integrateddynamics:menril_log", 0, logAmount, logChance);
    menrilTree.addDrop("integrateddynamics:menril_sapling", 0, saplingAmount, saplingChance);
    menrilTree.addDrop("integrateddynamics:menril_leaves", 0, leafAmount, leafChance);

    menrilTree.addDrop("integrateddynamics:crystalized_menril_chunk", 0, fruitAmount, fruitChance);
    menrilTree.addDrop("integrateddynamics:menril_berries", 0, fruitAmount, fruitChance);

    menrilTree.setGrowTime(defaultGrowTime);

    TreeTypeRegistry.registerTreeType(menrilTree);
};

File: src/main/java/org/dave/bonsaitrees/integration/mods/IntegratedDynamics.java
package org.dave.bonsaitrees.integration.mods;


import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import org.dave.bonsaitrees.api.*;

import java.util.Random;

import static org.dave.bonsaitrees.api.BonsaiDropChances.*;

@BonsaiIntegration(mod = "integrateddynamics")
public class IntegratedDynamics implements IBonsaiIntegration {
    @Override
    public void registerTrees(ITreeTypeRegistry registry) {
        TreeTypeSimple menrilTree = new TreeTypeSimple("integrateddynamics:menril", "integrateddynamics:menril_sapling", 0);

        menrilTree.addDrop(new ItemStack(Items.STICK, stickAmount), stickChance);

        menrilTree.addDrop("integrateddynamics:menril_log", logAmount, 0, logChance);
        menrilTree.addDrop("integrateddynamics:menril_sapling", saplingAmount, 0, saplingChance);
        menrilTree.addDrop("integrateddynamics:menril_leaves", leafAmount, 0, leafChance);

        menrilTree.addDrop("integrateddynamics:crystalized_menril_chunk", fruitAmount, 0, fruitChance);
        menrilTree.addDrop("integrateddynamics:menril_berries", fruitAmount, 0, fruitChance);

        registry.registerTreeType(this, menrilTree);
    }

    @Override
    public void generateTree(IBonsaiTreeType type, World world, BlockPos pos, Random rand) {

    }
}
